src/lib/rules/player.hh

```diff
@@ -33,38 +35,40 @@ struct Player
     std::string name;
 };
 
-using Player_sptr = std::shared_ptr<Player>;
-
-struct Players final : utils::IBufferizable
+class Players final : public utils::IBufferizable
 {
+    using PlayersVector = std::vector<std::shared_ptr<Player>>;
+
+public:
     Players() = default;
-    Players(std::vector<Player_sptr> players) : players(players) {}
+    Players(PlayersVector players) : players_{players} {}
 
     void handle_buffer(utils::Buffer& buf) override
     {
         if (buf.serialize())
         {
-            for (auto& p : players)
+            for (const auto& p : players_)
                 p->handle_buffer(buf);
         }
         else
         {
             while (!buf.empty())
             {
                 // Get a player
-                Player_sptr player = Player_sptr(new Player(0, 0));
+                auto player = std::make_shared<Player>();
 
                 // And unserialize it
                 player->handle_buffer(buf);
-                players.push_back(player);
+
+                players_.push_back(player);
```

Comment:
Kinda orthogonal to this change but this pushes back the players without clearing players_ first, is this intended?

Reply:
Added clear().